File: routeMap/app/src/main/java/ru/android/zheka/coreUI/ErrorDialog.java
package ru.android.zheka.coreUI;

import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;
import io.reactivex.Observable;
import io.reactivex.functions.Consumer;
import ru.android.zheka.gmapexample1.R;

public class ErrorDialog {
    private final IActivity view;
    protected DialogConfig config;

    public void showError(Throwable throwable, Consumer <Boolean> consumer) {
        // translation can be here
        config = DialogConfig.builder()
                .labelValue(view.getActivity ().getResources ().getString (R.string.errorDialog_windowTitle))
                .contentValue(throwable.getMessage ())
                .context(view.getContext ())
                .positiveConsumer (consumer)
                .layoutId (R.layout.dialog_error)
                .titleId (R.id.errorDialog_windowTitle)
                .contentId (R.id.errorDialog_value)
                .poistiveBtnId (R.string.ok_save_point)
                .build();
        show();
    }

    public ErrorDialog(DialogConfig config, IActivity view) {
        this.config = config;
        this.view = view;
    }

    public void show() {
        View view = getDialogView ();
        AlertDialog dialog = configureDialog (view);
        dialog.setCanceledOnTouchOutside (false);
        dialog.show ();
    }

    protected AlertDialog configureDialog(View view) {
        getContent (view);
        return new AlertDialog.Builder (config.context)
                .setView (view)
                .setPositiveButton (config.poistiveBtnId, (d,which) -> {
                    d.cancel ();
                    Observable.just(true).subscribe(config.positiveConsumer, this.view::showError).dispose();
                })
                .create ();
    }

    private View getDialogView() {
        View view = View.inflate (config.context, config.layoutId, null);
        TextView titleView = getDialogTitle (view);
        String value = config.labelValue;
        if (!value.isEmpty ()) {
            titleView.setText (value);
        }
        return view;
    }

    private TextView getDialogTitle(View view) {
        return view.findViewById (config.titleId);
    }

    protected TextView getContent(View view) {
        TextView content = view.findViewById (config.contentId);
        content.setText (config.contentValue);
        content.setOnClickListener (v -> content.setError (null));
        return content;
    }
}

File: routeMap/app/src/main/java/ru/android/zheka/coreUI/ErrorDialog.kt
package ru.android.zheka.coreUI

import android.content.DialogInterface
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import io.reactivex.Observable
import io.reactivex.functions.Consumer
import ru.android.zheka.gmapexample1.R

open class ErrorDialog(protected var config: DialogConfig, private val view: IActivity) {
    fun showError(throwable: Throwable, consumer: Consumer<Boolean>) {
        val isValidRes = view.activity != null && view.activity!!.resources != null
        // translation can be here
        config = DialogConfig.Companion.builder()
                .labelValue(if (isValidRes) view.activity.resources.getString(R.string.errorDialog_windowTitle) else "")
                .contentValue(if (throwable.message != null) throwable.message else throwable.toString())
                .context(view.activity)
                .positiveConsumer(consumer)
                .layoutId(R.layout.dialog_error)
                .titleId(R.id.errorDialog_windowTitle)
                .contentId(R.id.errorDialog_value)
                .poistiveBtnId(R.string.ok_save_point)
                .build()
        show()
    }

    fun show() {
        val view = dialogView
        val dialog = configureDialog(view)
        dialog.setCanceledOnTouchOutside(false)
        dialog.show()
    }

    protected open fun configureDialog(view: View): AlertDialog {
        getContent(view)
        return AlertDialog.Builder(config.context!!)
                .setView(view)
                .setPositiveButton(config.poistiveBtnId) { d: DialogInterface, which: Int ->
                    d.cancel()
                    Observable.just(true).subscribe(config.positiveConsumer, Consumer { throwable: Throwable -> this.view.showError(throwable) }).dispose()
                }
                .create()
    }

    private val dialogView: View
        private get() {
            val view = View.inflate(config.context, config.layoutId, null)
            val titleView = getDialogTitle(view)
            val value = config.labelValue
            if (!value!!.isEmpty()) {
                titleView.text = value
            }
            return view
        }

    private fun getDialogTitle(view: View): TextView {
        return view.findViewById(config.titleId)
    }

    protected fun getContent(view: View): TextView {
        val content = view.findViewById<TextView>(config.contentId)
        content.text = config.contentValue
        content.setOnClickListener { v: View? -> content.error = null }
        return content
    }

}